sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala

```diff
@@ -86,6 +93,429 @@ class DataFrameWindowFunctionsSuite extends QueryTest with SharedSQLContext {
     assert(e.message.contains("requires window to be ordered"))
   }
 
+  test("aggregation and rows between") {
+    val df = Seq((1, "1"), (2, "1"), (2, "2"), (1, "1"), (2, "2")).toDF("key", "value")
```

Comment:
We shall also include null data.

Reply:
This tests was removed and re-added as result of merge conflict. Now I cleaned up.